### ISSUE TITLE
[PMI-83]: Enable user provided schema

### ISSUE THREAD

**Issue by morazow** [EXASOL/spark-exasol-connector#5]
User provides a `schema` (StructType) to datasource with option `.option("schema", schemaStruct)` or `.schema(schemaStruct)` when reading so that the connector does not have to infer (jdbc call to Exasol) the schema of user provided query.

For this, DefaultSource should extend [SchemaRelationProvider](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala#L99-L111).

And override this function:
                                                                                 
```scala                                                                         
trait SchemaRelationProvider {                                                   
  /**                                                                            
   * Returns a new base relation with the given parameters and user defined schema.
   */                                                                            
  def createRelation(                                                            
      sqlContext: SQLContext,                                                    
      parameters: Map[String, String],                                           
      schema: StructType): BaseRelation                                          
}                                                                                
```

**Comment by 3cham** [comment]
Since I am working on this, one question: what if user provide both .option("schema",) and .schema(), should the one in .option() be preferred? I think it rarely happens but worth asking.

Without providing query, the table name must be provided as well. Should it be fully quallified (schema.table) or in two separate option?

**Comment by morazow** [comment]
Hey @3cham,

When I created this issue I did not think through, it is good that you are asking those questions!

No, let's not provide schema with an `.option`. The preferred api is to use `.schema` from Spark side also. Additionally, with option we will have to convert `StructType` to string and back because parameters is of type `Map[String, String]`. Long story short, ignore schema in option, let's use `.schema` and create relation in `def createRelation(..., schema:StructType): ..` function.

For your second point, let us not do anything yet. If we do not have a `query` string in one of options we fail fast. Users can put generic `select * from schema.table` in there if they need whole table.
